source/_docs/automation/templating.markdown

```diff
@@ -13,18 +13,20 @@ The variable `this` is the [state object](/docs/configuration/state_object) of t
 
 ## Available Trigger Data
 
-The following tables show the available trigger data per platform.
+The variable `trigger` is an object that contains details about which trigger activated the automation.  The data can be used by templates to modify the actions that are performed by the automation or displayed in a message.  For example, you could create an automation that that can be triggered by multiple sensors and then use the location of the sensor to specify a light to activate; or you could send a notification that contains the friendly name of the sensor that triggered it.
+
+Each [trigger platform](/docs/automation/trigger/#event-trigger) can include additional data specific to that platform.
 
 ### All
 
-The following describes trigger data associated with all platforms.
+Triggers from all platform will include the following data.
 
 | Template variable | Data |
 | ---- | ---- |
-| `trigger.id` | Optional trigger `id`, or index of the trigger.
+| `trigger.id` | The [`id` of the trigger](/docs/automation/trigger/#trigger-id).  If `id` is omitted, this will be set to the trigger index.
```

Comment:
This text is confusing, as this is an event result: How would one omit this here?
Let's avoid the term `omitted` in this case to avoid confusion.

Reply:
@frenck - I agree in this context it's a bit confusing.  It is copied from [the existing Trigger ID documentation](https://www.home-assistant.io/docs/automation/trigger/#trigger-id) almost verbatim.  I didn't like the fact that the old docs which don't really make it clear how it works.

Do you have a suggestion for a better way?  What If I just drop the whole second sentence and keep the link to the Trigger ID page?

Reply:
Yup, like I said, I agree that it's a bit confusing.  Do you just want me to revert this change, or do you have a suggestion for a better alternative?